mkdocs/docs/tutorials/traffic-engineering/session-affinity.md

```diff
@@ -0,0 +1,179 @@
+            route: # matching users will always go to v2
+            - destination:
+                host: flowers-predictor-default.ns-candidate.svc.cluster.local
+              headers:
+                request:
+                  set:
+                    Host: flowers-predictor-default.ns-candidate
+                response:
+                  set:
+                    version: flowers-v2
+          - route: # non-matching users will always go to v1
+            - destination:
+                host: flowers-predictor-default.ns-baseline.svc.cluster.local
+              headers:
+                request:
+                  set:
+                    Host: flowers-predictor-default.ns-baseline
+                response:
+                  set:
+                    version: flowers-v1
+        ```
+
+## 5. Generate requests
+=== "KFServing"
+    Generate requests to your model as follows.
+ 
+    === "Port forward (terminal one)"
+        ```shell
+        INGRESS_GATEWAY_SERVICE=$(kubectl get svc -n istio-system --selector="app=istio-ingressgateway" --output jsonpath='{.items[0].metadata.name}')
+        kubectl port-forward -n istio-system svc/${INGRESS_GATEWAY_SERVICE} 8080:80
+        ```
+
+    === "Baseline requests (terminal two)"
+        ```shell
+        curl -o /tmp/input.json https://raw.githubusercontent.com/kubeflow/kfserving/master/docs/samples/v1beta1/rollout/input.json
+        while true; do
+        curl -v -H "Host: example.com" -H "userhash: 1111100000" localhost:8080/v1/models/flowers:predict -d @/tmp/input.json
+        sleep 0.29
+        done
+        ```
+
+    === "Candidate requests (terminal three)"
+        ```shell
+        curl -o /tmp/input.json https://raw.githubusercontent.com/kubeflow/kfserving/master/docs/samples/v1beta1/rollout/input.json
+        while true; do
+        curl -v -H "Host: example.com" -H "userhash: 1010101010" localhost:8080/v1/models/flowers:predict -d @/tmp/input.json
+        sleep 2.0
+        done
+        ```
+
+## 6. Define metrics
+Please follow [Step 6 of the quick start tutorial](../../../getting-started/quick-start/#6-define-metrics).
+
+## 7. Launch experiment
+=== "KFServing"
+
+    ```shell
+    kubectl apply -f $ITER8/samples/kfserving/session-affinity/experiment.yaml
+    ```
+
+    ??? info "Look inside experiment.yaml"
+        ```yaml linenums="1"
+        apiVersion: iter8.tools/v2alpha2
+        kind: Experiment
+        metadata:
+          name: session-affinity-exp
+        spec:
+          target: flowers
+          strategy:
+            testingPattern: A/B
+            deploymentPattern: FixedSplit
+            actions:
+              # when the experiment completes, promote the winning version using kubectl apply
+              finish:
+              - task: common/exec
+                with:
+                  cmd: /bin/bash
+                  args: [ "-c", "kubectl apply -f {{ .promote }}" ]
```

Comment:
This is incorrect. And this needs to corrected in other tutorials as well.  It should include _-n default_ (or some other way to reference the experiment namespace such as _{{ .this.metadata.namespace }}_ -- perhaps a future approach!).  Currently, we are creating these virtual services in the ite8-system namespace instead of the experiment namespace. We are not changing the one in the experiment namespace at all.

Reply:
Not true.

`kubectl apply -f {{ .promote }}` creates the virtual service **already** in the default namespace. This is the behavior of `kubectl` when no namespace is specified as part of the command or within the yaml.

Reply:
I did not want to introduce an "experiment namespace"... this enables me to keep instructions and tutorial and artifacts simple.

Reply:
> Not true. iter8-system namespace does not enter the picture here.
 
You are right. I am not sure what I was thinking and cannot reproduce what I think I was seeing.